src/core/device.c

```diff
@@ -766,7 +773,29 @@ static int device_dispatch_io(sd_event_source *source, int fd, uint32_t revents,
                 return 0;
         }
 
-        if (streq(action, "remove"))  {
+        if (streq(action, "change"))  {
+                _cleanup_free_ char *e = NULL;
+                Unit *u;
+
+
+                r = unit_name_from_path(sysfs, ".device", &e);
+                if (r < 0)
+                        return log_error_errno(r, "Failed to generate unit name from device path: %m");
+
+                u = manager_get_unit(m, e);
+                if (!u) {
+                        log_error("Failed to get unit from sysfs name.");
+                        return 0;
+                }
+
+                r = manager_add_job(m, JOB_RELOAD, u, JOB_REPLACE, NULL, NULL);
+                if (r < 0) {
+                        log_error("Failed to add job to manager.");
+                        return 0;
+                }
+                manager_dispatch_run_queue(m);
```

Comment:
That's what I tried to explain in the PR comment... i'm pretty sure that's not how you are supposed to do it, but i didn't find out the proper way... help welcomed.

What I figured out so far is that the manager main loop goes like this

* GC useless jobs in the queue
* other stuff
* wait for a sd_event
  * if the event was "a job needs to be done" : a "while there is a job to do" loop is started. 
    if new job are created inside this loop they will be run within this loop
  * if the event was "a udev event" the event is handled and the main manager loop is restarte
    jobs created here will be "installed" but not run.

The problem arises i the GC logic. an installed job that is not wanted by any other job will be cleaned up by the GC logic before it has a chance to run. 
I have observed that the ReloadPropagated jobs were executed properly but not the actuel (empty) device-reload job. It seems that some other  job manage to work in that logic, but I didn't figure out how. I'd gladly take any pointer here...

I'll look into your idea of propagating non-existant reload jobs too, so V2 of the patch will probably be quite different. Thx for  the review